### PR TITLE
fix(v3proxy:error): updates to error handling

### PR DESCRIPTION
**Pull request by kschelonka**
Add more verbose logging for better deugging.

Consolidate async error handling patterns:
Allow all errors to bubble up to the route handler,
and wrap the route handler with a method which catches
rejected promises and sends them to error handling
middleware (otherwise resolve the function).

This might not be any different than the existing method
where the next calls are embedded in multiple try/catch
blocks of the async route handler, but I think this way
is less prone to developer error. I'm also wondering if
it's possible that the local test methods resolve promises
slightly differently than when the application is deployed?
Because we're seeing some unexpected unhandled errors
in Sentry, when no errors should be unhandled.

[POCKET-10354]

[POCKET-10354]: https://mozilla-hub.atlassian.net/browse/POCKET-10354?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: servers/v3-proxy-api/src/middleware/errorHandlers.ts [hunk 1]
```diff
@@ -58,13 +58,14 @@ export function clientErrorHandler(
...
   const defaultMessage = 'Something Went Wrong';
+  const defaultHeaders = customErrorHeaders('INTERNAL_SERVER_ERROR');
   if (err instanceof ClientError) {
     res.status(err.response.status);
     // There might be more than 1 error, but we will just return the first
     const primaryError = err.response.errors?.[0];
-    const headers = customErrorHeaders(primaryError?.extensions.code);
-    // Set headers if they're not undefined
-    headers && res.set(headers);
+    const headers =
```
Comment (kschelonka) [inline servers/v3-proxy-api/src/middleware/errorHandlers.ts:66]: set default headers if there is no mapping

File: servers/v3-proxy-api/src/middleware/errorHandlers.ts [hunk 2]
```diff
@@ -96,18 +94,23 @@ export function logAndCaptureErrors(
...
       return next(err);
     }
   }
   // Okay, now we actually log things
-  serverLogger.error(`${req.method}: ${req.baseUrl + req.path}: ${err}`);
-  const query = { ...req.query };
-  delete query['access_token'];
-  Sentry.addBreadcrumb({
-    data: { body: req.body, query },
+  serverLogger.error({
```
Comment (kschelonka) [inline servers/v3-proxy-api/src/middleware/errorHandlers.ts:108]: more verbose and json-formatted errors

File: servers/v3-proxy-api/src/routes/v3Add.ts [hunk 3]
```diff
@@ -28,28 +29,33 @@ const v3AddController = async (
...
-    const accessToken = (data.access_token as string) ?? null;
-    const consumerKey = (data.consumer_key as string) ?? null;
-    const client = getClient(accessToken, consumerKey, headers);
-    const graphResponse = await processV3Add(client, variables, data.tags);
-    return res.json(graphResponse);
-  } catch (err) {
-    // Pass along to error handling middleware
-    // Has to be in a try/catch block due to async call
-    return next(err);
+    throw new InputValidationError(result.array({ onlyFirstError: true })[0]);
```
Comment (kschelonka) [inline servers/v3-proxy-api/src/routes/v3Add.ts:32]: reworking these methods to throw or return

File: servers/v3-proxy-api/src/routes/v3Add.ts [hunk 4]
```diff
@@ -28,28 +29,33 @@ const v3AddController = async (
...
+  const graphResponse = await processV3Add(client, variables, data.tags);
+  return graphResponse;
 };
 
-router.get('/', checkSchema(V3AddSchema, ['query']), v3AddController);
-router.post('/', checkSchema(V3AddSchema, ['body']), v3AddController);
+router.get(
+  '/',
+  checkSchema(V3AddSchema, ['query']),
+  asyncHandler(async (req, res, next) => {
```
Comment (kschelonka) [inline servers/v3-proxy-api/src/routes/v3Add.ts:46]: wrap the method in asyncHandler -- if throws for any reason, will be captured and sent along to error handling middleware

File: servers/v3-proxy-api/src/middleware/asyncHandler.ts [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+
+/**
+ * Wrap async middleware to call next(err) if the promise is rejected.
+ * Prior to express 5, errors returned from asynchronous functions
+ * invoked by route handler and middleware must be passed directly
+ * to the next function.
+ * @param fn the request handler middleware
+ * @returns wrapped middleware function with promise handling
+ */
+export const asyncHandler =
```
Comment (kschelonka) [inline servers/v3-proxy-api/src/middleware/asyncHandler.ts:11]: single source of truth wrapper for handling async methods which might throw

**Review by collectedmind — APPROVED** [review]

**Pull request merged**